server/src/main/java/org/elasticsearch/index/mapper/DataStreamTimestampFieldMapper.java

```diff
@@ -139,7 +139,7 @@ public void doValidate(MappingLookup lookup) {
         }
 
         DateFieldMapper dateFieldMapper = (DateFieldMapper) mapper;
-        if (dateFieldMapper.fieldType().isIndexed() == false) {
+        if (dateFieldMapper.fieldType().isIndexed() == false && dateFieldMapper.fieldType().hasDocValuesSkipper() == false) {
```

Comment:
This is necessary to account for the fact that we disable the inverted index on `@timestamp` replacing it with a doc values sparse index for LogsDB.